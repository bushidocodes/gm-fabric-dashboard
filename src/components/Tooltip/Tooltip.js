import React from "react";
import PropTypes from "prop-types";
import TooltipContent from "./components/TooltipContent";
import TooltipWrap from "./components/TooltipWrap";

export default function Tooltip({
  children,
  content,
  position,
  disabled,
  containerStyle = {}
}) {
  return (
    <TooltipWrap style={containerStyle} disabled={disabled}>
      {children}
      <TooltipContent position={position}>{content}</TooltipContent>
    </TooltipWrap>
  );
}

Tooltip.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.node,
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  content: PropTypes.string.isRequired,
  position: PropTypes.oneOf(["top", "bottom", "left", "right"])
};

Tooltip.defaultProps = {
  position: "bottom"
};
