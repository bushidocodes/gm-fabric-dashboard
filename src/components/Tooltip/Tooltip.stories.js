import React from "react";
import { storiesOf } from "@storybook/react";
import {
  withKnobs,
  text,
  select,
  boolean,
  object
} from "@storybook/addon-knobs/react";

import Tooltip from "./index.js";

const wrapperStyle = {
  display: "flex",
  width: "100vw",
  justifyContent: "center",
  height: "100vh",
  alignItems: "center"
};

storiesOf("Tooltip", module)
  .addDecorator(withKnobs)
  .add("default", () => {
    let content = text("Tooltip Content", "This is a super cool tooltip!");
    let position = select(
      "Position",
      ["top", "right", "bottom", "left"],
      "top"
    );
    return (
      <div style={wrapperStyle}>
        <Tooltip
          content={content}
          position={position}
          disabled={boolean("disabled", false)}
          containerStyle={object("containerStyle", {})}
        >
          Hover over me
        </Tooltip>
      </div>
    );
  });
