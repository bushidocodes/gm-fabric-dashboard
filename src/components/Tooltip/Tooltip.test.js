import React from "react";
import { render } from "enzyme";

import Tooltip from "./Tooltip";

function renderTooltip(position) {
  return render(
    <Tooltip content="Hello!" position={position}>
      Hover over me
    </Tooltip>
  );
}

describe("Tooltip", () => {
  let wrapper = renderTooltip("top");

  it("correctly positions tooltip content based on position prop", () => {
    expect(wrapper).toMatchSnapshot();
    wrapper = renderTooltip("bottom");
    expect(wrapper).toMatchSnapshot();
    wrapper = renderTooltip("left");
    expect(wrapper).toMatchSnapshot();
    wrapper = renderTooltip("right");
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a tooltip div with the correct content", () => {
    expect(wrapper.html().includes("Hello!")).toBe(true);
  });
});
