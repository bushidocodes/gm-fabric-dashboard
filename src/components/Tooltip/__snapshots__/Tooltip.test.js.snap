// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip correctly positions tooltip content based on position prop 1`] = `
.c1 {
  background-color: rgba(0,0,0,0.8);
  border-radius: 6px;
  color: rgb(255,255,255);
  font-family: NunitoSans,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 11px;
  opacity: 0;
  padding: 10px;
  position: absolute;
  text-align: left;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
  visibility: hidden;
  white-space: normal;
  width: 160px;
  z-index: 1070;
  bottom: 150%;
  left: 50%;
  margin-left: -80px;
}

.c0 {
  border-bottom: 1px solid rgb(217,217,217);
  cursor: help;
  display: inline-block;
  position: relative;
}

.c0:hover > :last-child {
  opacity: 1;
  visibility: visible;
  z-index: 1070;
}

<div
  class="c0"
>
  Hover over me
  <div
    class="c1"
  >
    Hello!
  </div>
</div>
`;

exports[`Tooltip correctly positions tooltip content based on position prop 2`] = `
.c1 {
  background-color: rgba(0,0,0,0.8);
  border-radius: 6px;
  color: rgb(255,255,255);
  font-family: NunitoSans,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 11px;
  opacity: 0;
  padding: 10px;
  position: absolute;
  text-align: left;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
  visibility: hidden;
  white-space: normal;
  width: 160px;
  z-index: 1070;
  top: 150%;
  left: 50%;
  margin-left: -80px;
}

.c0 {
  border-bottom: 1px solid rgb(217,217,217);
  cursor: help;
  display: inline-block;
  position: relative;
}

.c0:hover > :last-child {
  opacity: 1;
  visibility: visible;
  z-index: 1070;
}

<div
  class="c0"
>
  Hover over me
  <div
    class="c1"
  >
    Hello!
  </div>
</div>
`;

exports[`Tooltip correctly positions tooltip content based on position prop 3`] = `
.c1 {
  background-color: rgba(0,0,0,0.8);
  border-radius: 6px;
  color: rgb(255,255,255);
  font-family: NunitoSans,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 11px;
  opacity: 0;
  padding: 10px;
  position: absolute;
  text-align: left;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
  visibility: hidden;
  white-space: normal;
  width: 160px;
  z-index: 1070;
  right: 120%;
  top: -5px;
}

.c0 {
  border-bottom: 1px solid rgb(217,217,217);
  cursor: help;
  display: inline-block;
  position: relative;
}

.c0:hover > :last-child {
  opacity: 1;
  visibility: visible;
  z-index: 1070;
}

<div
  class="c0"
>
  Hover over me
  <div
    class="c1"
  >
    Hello!
  </div>
</div>
`;

exports[`Tooltip correctly positions tooltip content based on position prop 4`] = `
.c1 {
  background-color: rgba(0,0,0,0.8);
  border-radius: 6px;
  color: rgb(255,255,255);
  font-family: NunitoSans,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 11px;
  opacity: 0;
  padding: 10px;
  position: absolute;
  text-align: left;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
  visibility: hidden;
  white-space: normal;
  width: 160px;
  z-index: 1070;
  top: -5px;
  left: 120%;
}

.c0 {
  border-bottom: 1px solid rgb(217,217,217);
  cursor: help;
  display: inline-block;
  position: relative;
}

.c0:hover > :last-child {
  opacity: 1;
  visibility: visible;
  z-index: 1070;
}

<div
  class="c0"
>
  Hover over me
  <div
    class="c1"
  >
    Hello!
  </div>
</div>
`;
